src/image/Exposure.cc

```diff
@@ -249,15 +225,20 @@ Exposure<ImageT, MaskT, VarianceT> Exposure<ImageT, MaskT, VarianceT>::getCutout
     if (!hasWcs()) {
         throw LSST_EXCEPT(pex::exceptions::LogicError, "Cannot look up source position without WCS.");
     }
-
     lsst::geom::Point2D pixelCenter = getWcs()->skyToPixel(center);
+
     if (!lsst::geom::Box2D(getBBox()).contains(pixelCenter)) {
         std::stringstream buffer;
         buffer << "Point " << center << " lies at pixel " << pixelCenter << ", which lies outside Exposure "
                << getBBox();
         throw LSST_EXCEPT(pex::exceptions::InvalidParameterError, buffer.str());
     }
-    lsst::geom::Box2I bbox = _makeCenteredBox(pixelCenter, size);
+    if (size[0] <= 0 || size[1] <= 0) {
+        std::stringstream buffer;
+        buffer << "Cannot create bounding box with dimensions " << size;
+        throw LSST_EXCEPT(pex::exceptions::InvalidParameterError, buffer.str());
+    }
+    lsst::geom::Box2I bbox = lsst::geom::Box2I::makeCenteredBox(pixelCenter, size, false);
```

Comment:
I don't see a good way to simplify this, though I'm also not too bothered by that - the overall code complexity is the same; you've just moved this check from one place to another.

But it may be worth noting that given the check for both the center coordinate being in within the Exposure bbox and the size being positive, I believe it's guaranteed that `overlapBox.isEmpty()` is always `false` up in `_copyCommonPixels`, and could perhaps be turned into an `assert`, *if* this is the only caller.

Reply:
I'd prefer to leave it in, in case somebody later wants to promote `_copyCommonPixels` to a public method or function. I had quite a bit of trouble generalizing `makeCenteredBox` to work correctly with negative boxes, and `assign`'s behavior is pretty surprising.

Reply:
> `assign`'s behavior is pretty surprising

I had no idea that's how it worked until I saw your comment, and now that I do know, it makes me sad, too.